### ISSUE TITLE
Add to android project

### ISSUE THREAD

**Issue by Zaherkorechi** [thoebert/krosbridge#1]
Hello @thoebert,

Thanks for the code. I am trying to add it to test project. But I am getting this error:

`An exception occurred applying plugin request [id: 'io.github.thoebert.krosbridge-codegen', version: '1.0.2']
> Failed to apply plugin 'io.github.thoebert.krosbridge-codegen'.
   > SourceSet with name 'main' not found.`

I am wondaring if I missed any things, this my build.gradle.kts file code
`//import com.github.thoebert.krosbridgecodegen.KROSBridgeCodegenPluginConfig

plugins {
    id("com.android.application")
    id("org.jetbrains.kotlin.android")
    kotlin("plugin.serialization") version "1.7.21"
    id("io.github.thoebert.krosbridge-codegen") version "1.0.2"
}

android {
    namespace = "com.example.testrosbridge"
    compileSdk = 33

    defaultConfig {
        applicationId = "com.example.testrosbridge"
        minSdk = 24
        targetSdk = 33
        versionCode = 1
        versionName = "1.0"

        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
        vectorDrawables {
            useSupportLibrary = true
        }
    }

    buildTypes {
        release {
            isMinifyEnabled = false
            proguardFiles(
                getDefaultProguardFile("proguard-android-optimize.txt"),
                "proguard-rules.pro"
            )
        }
    }
    compileOptions {
        sourceCompatibility = JavaVersion.VERSION_1_8
        targetCompatibility = JavaVersion.VERSION_1_8
    }
    kotlinOptions {
        jvmTarget = "1.8"
    }
    buildFeatures {
        compose = true
    }
    composeOptions {
        kotlinCompilerExtensionVersion = "1.4.3"
    }
    packaging {
        resources {
            excludes += "/META-INF/{AL2.0,LGPL2.1}"
        }
    }
}

dependencies {

    implementation("androidx.core:core-ktx:1.10.1")
    implementation("androidx.lifecycle:lifecycle-runtime-ktx:2.6.1")
    implementation("androidx.activity:activity-compose:1.7.2")
    implementation(platform("androidx.compose:compose-bom:2023.03.00"))

    // Required
    implementation("com.github.thoebert:krosbridge:main-SNAPSHOT")
    implementation("org.jetbrains.kotlinx:kotlinx-serialization-json:1.5.0")
    implementation("io.ktor:ktor-client-okhttp:2.2.4")

    // Optional for logging:
    implementation("org.apache.logging.log4j:log4j-core:2.17.1")
    implementation("org.apache.logging.log4j:log4j-slf4j-impl:2.17.1")

    implementation("androidx.compose.ui:ui")
    implementation("androidx.compose.ui:ui-graphics")
    implementation("androidx.compose.ui:ui-tooling-preview")
    implementation("androidx.compose.material3:material3")
    implementation("com.google.firebase:firebase-crashlytics-buildtools:2.9.9")
    testImplementation("junit:junit:4.13.2")
    androidTestImplementation("androidx.test.ext:junit:1.1.5")
    androidTestImplementation("androidx.test.espresso:espresso-core:3.5.1")
    androidTestImplementation(platform("androidx.compose:compose-bom:2023.03.00"))
    androidTestImplementation("androidx.compose.ui:ui-test-junit4")
    debugImplementation("androidx.compose.ui:ui-tooling")
    debugImplementation("androidx.compose.ui:ui-test-manifest")
}

//configure<KROSBridgeCodegenPluginConfig> {
//    packageName.set("com.company.project.messages") // package for generated data classes
//}`

Regarding the jitpack, I added it to settings.gradle.kts like this:

`pluginManagement {
    repositories {
        google()
        mavenCentral()
        gradlePluginPortal()
        maven(url = "https://jitpack.io")
    }
}
dependencyResolutionManagement {
    repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)
    repositories {
        google()
        mavenCentral()
    }
}`

**Comment by thoebert** [comment]
Hello!

So far I only implemented and used krosbridge on the JVM and not Android. But since you asked, I just added and tested Android support by updating my code generation plugin and resolved the missing sources of the android gradle plugin.

Just make sure to use the updated version 1.0.5 of the code-gen plugin and add the generated sources to the sourceset as documented in the updated Readme.

Thank you!

**Comment by Zaherkorechi** [comment]
Hello @thoebert,

Sorry, for the very late response. And thank you very much for helping.

I tried out again with the new update, but I am getting this error 
` ./gradlew generateROSSources
Starting a Gradle Daemon, 1 incompatible and 10 stopped Daemons could not be reused, use --status for details

FAILURE: Build failed with an exception.

* What went wrong:
The Android Gradle plugin supports only Kotlin Gradle plugin version 1.5.20 and higher.
The following dependencies do not satisfy the required version:
project ':app' -> io.github.thoebert.krosbridge-codegen:io.github.thoebert.krosbridge-codegen.gradle.plugin:1.0.5 -> io.github.thoebert:krosbridge-codegen:1.0.5 -> org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.61

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.`

I tried to downgrade the Kotlin version and gradle plugin to 7.2 as you used to build module but that didn't help. 
I don't know, what exactly I need to do to make it works.
